stdlib/public/Concurrency/Task+init.swift.gyb

```diff
@@ -0,0 +1,209 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2020 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import Swift
+
+// ==== Task.init ------------------------------------------------
```

Comment:
Continuing the quest to stop writing almost the same method 24 times 😉 

- init (3)
  - init 
  - SWIFT_STDLIB_TASK_TO_THREAD_MODEL_CONCURRENCY
  - $Embedded version
- init + throwing (3)
  - ...
- detached (3)
  - ...
- detached + throwing (3)
  - ...

So we're saving some repetition here... This should eventually also consume the `Task.swift` and `Task+TaskExecutor.swift` initializers, but I'm taking it one step at a time, to not cause unnecessary risk for this change.